### PR TITLE
Feat/counter and list defaulting order to ascending

### LINKED ISSUE #3683: Counter and List Priorities: Default to Ascending.
**Is your feature request related to a problem? Please describe.**

A small quality of life improvement - it would be really nice for `Fleet.Spec.Priorities` and `GameServerAllocation.Spec.Priorities` the default for `order` as `Ascending`.

**Describe the solution you'd like**

If not defined, make the order `Ascending`.

**Describe alternatives you've considered**

Leave it alone and let people be explicit with their ordering.

**Additional context**

We have the following in our CRD:

https://github.com/googleforgames/agones/blob/1c536c0aec680e8b013479bc15481c4ae47e3185/install/helm/agones/templates/crds/fleet.yaml#L129-L134

So at some point, someone figured we should do this 😄

### PR DESCRIPTION
**Pull request by lacroixthomas**
**What type of PR is this?**
/kind feature

**What this PR does / Why we need it**:
Defaulting the `order` from counter and list priorities to `Ascending`

**Which issue(s) this PR fixes**:

Closes #3683 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by markmandel** [comment]
@Kalaiselvi84 @lacroixthomas - just noticed you duplicated each other's work 😄 

I added some review to https://github.com/googleforgames/agones/pull/3736 on some tests , primarily on the unit test that needs expanding.

No strong opinions on my end who wraps this one up 😄

File: pkg/apis/allocation/v1/gameserverallocation.go [hunk 1]
```diff
@@ -583,6 +583,12 @@ func (gsa *GameServerAllocation) ApplyDefaults() {
 		gsa.Spec.Scheduling = apis.Packed
 	}
 
+	for i := range gsa.Spec.Priorities {
```
Comment (markmandel) [inline pkg/apis/allocation/v1/gameserverallocation.go:586]: Please add tests to:

https://github.com/googleforgames/agones/blob/a1a1cd956a19e4e97fd67f6c0ad06da4259273ab/pkg/apis/allocation/v1/gameserverallocation_test.go#L31

Comment (markmandel) [reply pkg/apis/allocation/v1/gameserverallocation.go:586]: Can we also please add a comment here:

https://github.com/googleforgames/agones/blob/986ed9b98f9e830e131f73c9347d17d1eb05b32f/pkg/apis/agones/v1/common.go#L186

That the default is `Ascending`.

**Review by markmandel — COMMENTED** [review]
Copying review across, and adding a few more things - but otherwise looks good 👍🏻

File: pkg/apis/allocation/v1/gameserverallocation.go (hunk 1 above)
Comment (lacroixthomas) [reply pkg/apis/allocation/v1/gameserverallocation.go:586]: MR updated following your comments @markmandel 🙂

**Review by markmandel — APPROVED** [review]
Awesome. Thanks!

**Pull request merged**